pandas/_libs/src/datetime/np_datetime.c

```diff
@@ -529,51 +513,34 @@ int convert_pydatetime_to_datetimestruct(PyObject *obj,
 
 npy_datetime pandas_datetimestruct_to_datetime(PANDAS_DATETIMEUNIT fr,
                                                pandas_datetimestruct *d) {
-    pandas_datetime_metadata meta;
     npy_datetime result = PANDAS_DATETIME_NAT;
 
-    meta.base = fr;
-    meta.num = 1;
-
-    convert_datetimestruct_to_datetime(&meta, d, &result);
+    convert_datetimestruct_to_datetime(fr, d, &result);
     return result;
 }
 
 void pandas_datetime_to_datetimestruct(npy_datetime val, PANDAS_DATETIMEUNIT fr,
                                        pandas_datetimestruct *result) {
-    pandas_datetime_metadata meta;
```

Comment:
could remove these functions entirely (and just call the convert_*) ones?

Reply:
Eventually.  There are still some calls from np_datetime_strings.c and ujson/python/objToJSON.c so it isn't entirely trivial.